src/components/Tabs/__tests__/TabsItem.test.tsx

```diff
@@ -0,0 +1,168 @@
+        const counter = 3;
+
+        render(<TabsItem id={tabId} title={tabTitle} counter={counter} onClick={jest.fn()} />);
+
+        const component = screen.getByRole('tab');
+        const counterComponent = screen.getByText(counter);
+
+        expect(counterComponent).toBeVisible();
+        expect(counterComponent).toHaveClass('yc-tabs__item-counter');
+        expect(component).toContainElement(counterComponent);
+    });
+
+    test('should render label', () => {
+        const labelQaId = 'label-qa-id';
+
+        const label = {
+            theme: 'normal' as const,
+            content: <span data-qa={labelQaId}>label content</span>,
+        };
+
+        render(<TabsItem id={tabId} title={tabTitle} label={label} onClick={jest.fn()} />);
+
+        const component = screen.getByRole('tab');
+        const labelComponent = screen.getByTestId(labelQaId);
+
+        expect(labelComponent).toBeVisible();
+        expect(component).toContainElement(labelComponent);
+    });
+
+    test('should render icon', () => {
+        const iconQaId = 'icon-qa-id';
+
+        const icon = <GearIcon data-qa={iconQaId} width={20} height={20} />;
+
+        render(<TabsItem id={tabId} title={tabTitle} icon={icon} onClick={jest.fn()} />);
+
+        const component = screen.getByRole('tab');
+        const iconComponent = screen.getByTestId(iconQaId);
+
+        expect(iconComponent).toBeVisible();
+        expect(component).toContainElement(iconComponent);
+    });
+
+    test('should render meta', () => {
+        const meta = 'meta content';
+
+        render(<TabsItem id={tabId} title={tabTitle} meta={meta} onClick={jest.fn()} />);
+
+        const component = screen.getByRole('tab');
+        const metaComponent = screen.getByText(meta);
+
+        expect(metaComponent).toBeVisible();
+        expect(metaComponent).toHaveClass('yc-tabs__item-meta');
+        expect(component).toContainElement(metaComponent);
+    });
+
+    test('should call onClick on tab click', async () => {
+        const onClickFn = jest.fn();
+        const user = userEvent.setup();
+
+        render(<TabsItem id={tabId} title={tabTitle} onClick={onClickFn} />);
+
+        const component = screen.getByRole('tab');
+
+        await user.click(component);
+        expect(onClickFn).toBeCalledWith(tabId);
+    });
+
+    test('should call onClick on space click', async () => {
+        const onClickFn = jest.fn();
+
+        render(<TabsItem id={tabId} title={tabTitle} onClick={onClickFn} />);
+
+        const component = screen.getByRole('tab');
+
+        fireEvent.keyDown(component, {keyCode: 32});
+
+        expect(onClickFn).toBeCalledWith(tabId);
+    });
+
+    test('should not call onClick on tab click', async () => {
+        const onClickFn = jest.fn();
+
+        render(<TabsItem id={tabId} title={tabTitle} onClick={onClickFn} />);
+
+        const component = screen.getByRole('tab');
+
+        fireEvent.keyDown(component, {keyCode: 9});
```

Comment:
can we use `@testing-library/user-event`?

Reply:
done)